unstable/zwlr-screencontent-v1.xml

```diff
@@ -0,0 +1,209 @@
+      "receive_object" callback once per object or the "receive_layer" callback,
+      once per layer. The "receive_plane" callback shall only be called after
+      the "receive_layer" callback corresponding to the layer the plane belongs
+      to has been called. Finally, the "ready" event is called to indicate that
+      all the data has been made available for readout, as well as the time
+      at which presentation happened at.
+      The ownership of the frame is passed to the client, who's responsible for
+      destroying it once ready.
+      The data the API describes has been based off of what
+      VASurfaceAttribExternalBuffers contains.
+      All frames are read-only and may not be written into or altered.
+    </description>
+
+    <enum name="frame_type">
+      <entry name="byte_buffer" value="0"
+             summary="file descriptors hold mappable linear memfd buffers"/>
+      <entry name="dma_buf" value="1"
+             summary="file descriptors contain valid dmabuf objects"/>
+    </enum>
+
+    <enum name="frame_flags">
+      <description summary="frame flags">
+        Special flags that must be respected by the client.
+        Transient frames indicate short lifetime frames (such as swapchain
+        images from external clients). Clients are advised to copy them and do
+        all processing outside of the "ready" event.
+      </description>
+      <entry name="transient" value="1"/>
+    </enum>
+
+    <event name="receive">
+      <description summary="main callback">
+        Main callback supplying the client with information about the frame,
+        as well as an object to serve as context for destruction. Always called
+        first before any other events.
+      </description>
+      <arg name="width" type="uint"
+           summary="frame width, scaling factor included"/>
+      <arg name="height" type="uint"
+           summary="frame height, scaling factor included"/>
+      <arg name="scale" type="uint"
+           summary="current scaling factor of the client"/>
+      <arg name="type" type="uint" enum="frame_type"
+           summary="indicates the type of frame preferred"/>
+      <arg name="frame_flags" type="uint" enum="frame_flags"
+           summary="indicates special frame features"/>
+      <arg name="mod_high" type="uint"
+           summary="drm format modifier, high"/>
+      <arg name="mod_low" type="uint"
+           summary="drm format modifier, low"/>
+      <arg name="num_objects" type="uint"
+           summary="indicates how many objects (FDs) the frame has (max 4)"/>
+      <arg name="num_layers" type="uint"
+           summary="indicates how many layers the frame has (max 4)"/>
+    </event>
+    <event name="receive_object">
+      <description summary="object receiving callback">
+        Callback which serves to supply the client with the file descriptors
+        containing the data for each object.
+        The amount of data in each FD can be inferred via lseek().
```

Comment:
At least weston does not assume that dmabufs are seekable:

https://github.com/wayland-project/weston/blob/master/libweston/linux-dmabuf.c#L228

Reply:
Vulkan specifies that all mappable dmabuf file descriptors are seekable, but adding a field won't hurt. Done.